### PR TITLE
WIP: Parse str refactor

### PR DESCRIPTION
**Pull request by deg4uss3r**
At this point I'm not sure if this refactor will be a breaking change, happy to close this WIP: PR for opened against the breaking brance if you foresee this as a breaking change. 

**I'm submitting a(n)** refactor

# Description

Refactoring parse_str for code resuse in other fucntions/adapters

# Motivation

Issue #272  

# Tests

I am making sure all current tests pass first, then we can add more tests for the new modules if deemed necessary. 

# Related Issue(s)

#272

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -933,17 +933,15 @@ impl Uuid {
...
-    /// hyphens.
-    ///
-    /// Any of the formats generated by this module (simple, hyphenated, urn)
-    /// are supported by this parsing function.
-    pub fn parse_str(mut input: &str) -> Result<Uuid, parser::ParseError> {
-        // Ensure length is valid for any of the supported formats
+    /// Parses the length of a provided UUID and returns true if the
+    /// length matches a valid from (urn, hexadecimal, or hyphenated hex)
+    /// Returns valid UUID or ParseError
+    pub fn valid_length_check(input: &str) -> Result<&str, parser::ParseError> {
```
Comment (kinggoesgaming) [inline src/lib.rs]: move all parsing functions into the `parser` module as free functions

File: src/lib.rs [hunk 2]
```diff
@@ -933,17 +933,15 @@ impl Uuid {
...
-    /// Parses a `Uuid` from a string of hexadecimal digits with optional
-    /// hyphens.
-    ///
-    /// Any of the formats generated by this module (simple, hyphenated, urn)
-    /// are supported by this parsing function.
-    pub fn parse_str(mut input: &str) -> Result<Uuid, parser::ParseError> {
-        // Ensure length is valid for any of the supported formats
+    /// Parses the length of a provided UUID and returns true if the
+    /// length matches a valid from (urn, hexadecimal, or hyphenated hex)
+    /// Returns valid UUID or ParseError
```
Comment (kinggoesgaming) [inline src/lib.rs]: 1. UUID --> Uuid
2. Link the types

File: src/lib.rs [hunk 3]
```diff
@@ -933,17 +933,15 @@ impl Uuid {
...
     }
 
-    /// Parses a `Uuid` from a string of hexadecimal digits with optional
-    /// hyphens.
-    ///
-    /// Any of the formats generated by this module (simple, hyphenated, urn)
-    /// are supported by this parsing function.
-    pub fn parse_str(mut input: &str) -> Result<Uuid, parser::ParseError> {
-        // Ensure length is valid for any of the supported formats
+    /// Parses the length of a provided UUID and returns true if the
```
Comment (kinggoesgaming) [inline src/lib.rs]: Uuid and link

**Comment by deg4uss3r** [comment]
Fixed a few of the request, let me know how this looks. Sorry was at a wedding and today is a travel day! Hopefully will have some more time later :)

**Comment by Dylan-DPC** [comment]
@deg4uss3r Hi. Somehow you have messed up your updation process, it now shows  us all commits which are on master, instead of just your commit. Can you update the branch again?

**Comment by deg4uss3r** [comment]
Hey guys, I marked the requested changes resolved, but git appears to be stuck on that review?

I made one small change, and wanted to see what else you consider that could be extracted/useful for this PR.

**Comment by kinggoesgaming** [comment]
@deg4uss3r 

I think ill make a stub PR that introduces a set of functions in other structs with `unimplemented!()` so that it can help you with this.

**Comment by deg4uss3r** [comment]
Thanks @kinggoesgaming ! Hopefully I'm being helpful here, but I wanted to make sure anything I refactor makes sense to be used elsewhere in the code.

**Comment by kinggoesgaming** [comment]
Give me until after Monday to get that pr going (midterms season)

**Comment by kinggoesgaming** [comment]
This PR is horribly out of date and would be better off with a new PR.

@deg4uss3r you still willing to work on this? or we can get it going separately

**Comment by deg4uss3r** [comment]
Yeah, still willing to help out :)

**Pull request closed without merging**